app/models/query_parameter.rb

```diff
@@ -0,0 +1,16 @@
+class QueryParameter < ApplicationRecord
+
+  MODES = {mandatory: 1, optional: 2, forbidden: 3}
+  enum mode: MODES
+
+  belongs_to :route
+
+  validates :name, uniqueness: true, presence: true
```

Comment:
be careful with "uniqueness", because it will check for uniqueness in the entire schema of the client
if he got more than one route, query_parameters will be uniqueness on ALL routes
but we want the query_parameter to be uniqueness per route only